core/src/main/java/org/mule/api/el/ExpressionLanguage.java

```diff
@@ -109,4 +110,5 @@
     @Deprecated
     <T> T evaluate(String expression, MuleMessage message, Map<String, Object> vars);
 
+    TypedValue evaluateTyped(String expression, MuleMessage message);
```

Comment:
I am not sure but want to add some point here with my experience. When ever i use Z_ or C_ in expression then the evaluation not happening as expected is this relevant here? If not please Ignore